### ISSUE TITLE
CocoaPods framework inside a MacOS system extension

### ISSUE THREAD

**Issue by munibsiddiqui** [CocoaPods/CocoaPods#10486]
I'm trying to create an application that uses a system extension (NEFilterPacketProvider to be exact). Currently, I'm using CocoaPods to manage my dependencies. Everything works fine, up until the point when I try to use a framework inside my system extension.

After some investigation, it turns out that the framework is not included as part of system extension, but rather copied inside the main application bundle. System extension target simply references the relative framework directory of the application, as part of runtime framework search paths. This is problematic, because system extension is copied to /Library/SystemExtension/* after installation. Because the search path is relative, it no longer works, since extension has been copied outside the application.

After doing some Googling, it seems that the "way" of CocoaPods is to keep all of the frameworks inside application bundle. Ideally, I would like to have CocoaPods copy the framework into my system extension bundle. This would ensure that the system extension can operate as a standalone binary, without having to depend on application being present at a hardcoded path.

I'm wondering if there are any better solutions than just adding /Application/XXX.app/Contents/Frameworks to runtime framework search paths, and telling the user that the extension won't work otherwise.

Ref: https://stackoverflow.com/questions/64929562/how-to-include-cocoapods-framework-inside-a-macos-system-extension

## What did you do?
Install the following dependency for app and system extension. 

**Podfile** 
```
target 'SimpleFirewall' do
  # Comment the next line if you don't want to use dynamic frameworks
  use_frameworks!
pod 'Alamofire'
  # Pods for SimpleFirewall

end

target 'SimpleFirewallExtension' do
  # Comment the next line if you don't want to use dynamic frameworks
  use_frameworks!
pod 'Alamofire'
  # Pods for SimpleFirewallExtension

end
```

## What happened instead?

Framework embedded in application but didn't embed in system extension. 

## CocoaPods Environment

`
### Stack

```
   CocoaPods : 1.10.1
        Ruby : ruby 2.6.3p62 (2019-04-16 revision 67580) [universal.x86_64-darwin20]
    RubyGems : 3.0.3
        Host : macOS 11.1 (20C69)
       Xcode : 12.4 (12D4e)
         Git : git version 2.24.3 (Apple Git-128)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib
Repositories : cocoapods - git - https://github.com/CocoaPods/Specs.git @ 3c73f7492e20c6ed9894b2bc70edccc715b60666

               trunk - CDN - https://cdn.cocoapods.org/
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.4
cocoapods-keys        : 2.2.0
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.1.0
cocoapods-trunk       : 1.4.1
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'SimpleFirewall' do
  # Comment the next line if you don't want to use dynamic frameworks
  use_frameworks!
pod 'Alamofire'
  # Pods for SimpleFirewall

end

target 'SimpleFirewallExtension' do
  # Comment the next line if you don't want to use dynamic frameworks
  use_frameworks!
pod 'Alamofire'
  # Pods for SimpleFirewallExtension

end
```

`

## Project that demonstrates the issue

[FilteringNetworkTraffic-Demo.zip](https://github.com/CocoaPods/CocoaPods/files/6077243/FilteringNetworkTraffic-Demo.zip)

**Comment by dnkoutso** [comment]
thanks for the issue and sample app! we will have to investigate

**Comment by dnkoutso** [comment]
@munibsiddiqui can macOS system extensions be ran independently and without being embedded to an app?

**Comment by munibsiddiqui** [comment]
@dnkoutso Yes thats correct. macOS system extensions can run independently without embedded to an app.
